runtime/service_integ_test.go

```diff
@@ -485,12 +484,16 @@ func TestStubBlockDevices_Isolated(t *testing.T) {
 	})
 	require.NoError(t, err, "failed to create VM")
 
+	script, err := ioutil.ReadFile("/tmp/scripts/lsblk.sh")
+	require.NoError(t, err)
+
 	newContainer, err := client.NewContainer(ctx,
 		containerName,
 		containerd.WithSnapshotter(naiveSnapshotterName),
 		containerd.WithNewSnapshot(snapshotName, image),
 		containerd.WithNewSpec(
-			oci.WithProcessArgs("lsblk"),
+			oci.WithEnv([]string{"SCRIPT=" + string(script)}),
```

Comment:
Do you mind trying out quickly adding `oci.WithParentCgroupDevices` and `oci.WithCgroup("")` here and seeing if that allows you to actually read the block devices? https://github.com/containerd/containerd/blob/41e172352c41b6ff9e3503241d1bb17bfbcc28a0/oci/spec_opts.go#L988

`WithParentCgroupDevices` is an option not available in the normal `ctr` command line and on the surface seems like it in combination with `WithCgroup` might do what we want. If it doesn't work, no worries, but I think it's worth a shot as the test is significantly improved if we can verify each stub drive is being written with the right contents

Reply:
Thanks. WithParentCgroupDevices works! https://github.com/firecracker-microvm/firecracker-containerd/pull/233/commits/569fea02d0d241a3f85e4ba661e484b0a2d21e1f

By default, a containerd client passes `LinuxDeviceCgroup { Allow:false, Access: rwm }` (ref https://github.com/containerd/containerd/blob/7ff23438f64cfa62651ee9beea83f4dddfcf5a1e/oci/spec.go#L231). `WithParentCgroupDevices` clears this thing which makes /dev accessible.